vignettes/site_only/db_connections.Rmd

```diff
@@ -0,0 +1,183 @@
+
+```{r, eval=FALSE}
+library(healthcareai)
+library(DBI)
+
+my_con <- build_connection_string(server = "HCS-GM0004",
+                                  database = "SAM")
+con <- DBI::dbConnect(odbc::odbc(), .connection_string = my_con)
+```
+
+
+#### Using SQL Code
+
+From here, you can read data from the database in a couple different ways. You
+can use `db_read` to execute SQL code, or you can use `dbplyr` functionaliity
+to execute `dplyr` code directly against the database.
+```{r, eval=FALSE}
+query <- "SELECT
+          ,COUNT(*),
+          ,year(AdmitDT) as AdmitYear
+          FROM [SAM].[Encounter].[PatientEncounterBASE]
+          WHERE AdmitYear >= 2013
+          GROUP BY AdmitYear
+          ORDER BY AdmitYear"
+
+d <- db_read(con, query)
+```
+This query counts admits by year. If you don't want to pull the results into
+memory, as might the be the case with large tables, set the `pull_into_memory`
+flag to FALSE. This will create a pointer to the database that can be 
+executed "lazily." In other words R will wait until the last moment to execute 
+the statement.
+
+#### Using dplyr Code
+
+The other option is to use dplyr code. Set up a reference to the table, then use
+it to pull data. The last line, `collect()`, is what actually pulls the data 
+into memory.
+```{r, eval=FALSE}
+library(tidyverse)
+
+encounter <- tbl(con, in_schema("Encounter", "PatientEncounterBASE"))
+
+d <- encounter %>%
+  mutate(AdmitYear = year(AdmitDT)) %>%
+  filter(AdmitYear >= 2013) %>%
+  group_by(AdmitYear) %>%
+  tally() %>%
+  collect()
+```
+
+# Writing Data
+As of 4/1/2018, there are two ways to write to a database: 
+
+1. The latest version of `odbc`, and `DBI`, 
+installed from Github. The CRAN version does not support database schemas.
+2. `RODBC`
+
+In either case, the data you are trying to write must match the destination
+table's column names and data types.
+
+#### Writing with DBI
+
+DBI requires that you set up the table as an object before trying to write to
+it. Assuming there was a table called `SAM.Sepsis.Predictions` with columns:
+
+- BindingID
+- BindingNM
+- LastLoadDTS
+- patient_id (int)
+- predicted_probability (float).
+```{r, eval=FALSE}
+library(DBI)
+devtools::install_github("r-dbi/odbc")
+
+predictions <- data.frame(patient_id = c(1,2,3),
+                          predicted_readmission = c(0.7, 0.2, 0.4)) %>%
+  add_SAM_utility_cols()
+
+table_id <- Id(schema = "Sepsis", 
+                    name = "Predictions")
+
+res <- dbWriteTable(conn = my_con,
+                         name = table_id,
+                         value = predictions,
+                         append = TRUE)
+dbDisconnect(con)
+```
+
+#### Writing with RODBC
+
+RODBC is not the preferred way to interact with a database because it doesn't
+work with RStudio's connections pane. However, the package is available on CRAN
+and does support schemas. To use it, you must set up an RODBC connection to the 
+database, as a DBI connection won't work.
```

Comment:
@mmastand what benefit does `RODBC` provide here? Is the lack of CRAN schema support the only reason not to simplify and recommend folks use DBI/odbc?

Reply:
Yes. The lack of CRAN schema support is the only reason to not use. DBI.